src/main/java/com/alibaba/com/caucho/hessian/io/BasicSerializer.java

```diff
@@ -106,6 +106,8 @@ public void writeObject(Object obj, AbstractHessianOutput out)
                 break;
 
             case FLOAT:
+                out.writeDouble(Double.parseDouble(String.valueOf(((Number) obj).floatValue())));
```

Comment:
Is `out.writeFloat(((Number) obj).floatValue())` not ok?

Reply:
There is no such method

Reply:
Then how about `out.writeDouble(((Number) obj).floatValue())` ? I am concerning the float->String->double conversion, the intermediate String conversion should be avoid.

Reply:
When single precision is double precision, double precision will complement the single precision. Lead to deviation. 
For example:
99.8F  ->  99.80000305175781

Reply:
Okay, I got what you mean. It actually is:

```
jshell> float f = 99.8f
f ==> 99.8

jshell> double d = f;
d ==> 99.80000305175781
```

I do not have permission to merge this. I will approve this change.
This repo should be transfer to ASF soon. You should also have permission.